engine/swarm/store_sensitive_strings.md

```diff
@@ -0,0 +1,234 @@
+  such as Amazon AWS, EC2, or S3, and you want to prevent unauthorized use of
+  this infrastructure.
+- You are a developer building an application which processes sensitive data
+  to which you do not need or want direct access. Your administrator may
+  populate the secret key with a mock value while you are developing the
+  application, and change it to use the real value in production.
+
+In these and many other situations, managing sensitive strings using Docker
+secrets makes sense.
+
+## How Docker manages secrets
+
+When you add a secret to the swarm, Docker sends the secret to each manager
+node over a network connection which uses mutual TLS. When a manager receives
+a secret, it encrypts it with an encryption key before writing it to
+the RAFT logs, just as other RAFT log data is encrypted. The encryption key
+itself has already been propagated to the swarm node using mutual TLS.
+
+>**Warning**: RAFT data is encrypted in Docker 1.13 and newer. If some of your
+Swarm nodes run an earlier version, the secrets are stored unecrypted in
+those nodes' RAFT logs.
+
+When you create a service, you can give the service access to one or more
+secrets. You can update a service to grant it access to additional secrets or
+revoke its access to a given secret at any time.
+
+The Engine managing a given worker node creates a bind mount within
+`/run/secrets/` on the worker node for each secret. From the point of view of
+the container, the secret is available as plain text. The `/run/secrets/`
+directory is an in-memory filesystem. When the node stops or leaves the swarm,
+or when the service is stopped or removed, the secrets are no longer accessible.
+If a node loses connectivity to the swarm, it still has access to its secrets,
+but cannot receive updates until it reconnects to the swarm.
+
+You can add, remove, or get an individual secret at any time, or list all
+secrets. You can update a secret by removing it and creating a new secret with
+the same name. However, you cannot remove a secret that a running service is
+using. In order to update or roll back a secret, consider adding a version
+number or date to the secret name.
+
+## Read more about `docker secret` commands
+
+Use these links to read about specific commands, or continue to the
+[example about using secrets with a service](store_sensitive_strings.md@example-use-secrets-with-a-service).
+
+- [`docker secret create`](../reference/commandline/secret_create.md)
+- [`docker secret inspect`](../reference/commandline/secret_inspect.md)
+- [`docker service ls`](../reference/commandline/secret_ls.md)
+- [`docker secret rm`](../reference/commandline/secret_rm.md)
+- [`--secret`](../reference/commandline/service_create.md#create-a-service-with-secrets) flag for `docker service create`
+- [`--secret-add` and `--secret-rm`](../reference/commandline/service_update.md#adding-and-removing-secrets) flags for `docker service update`
+
+## Example: Use secrets with a service
```

Comment:
We need an example for rotation of a secret (add a new secret, remove the old one, and ensure the new secret has the same target as the old one.

We should probably also give advanced examples on how to change the permissions of the secret file, and the use of `source=bla,target=bla,uid=1000,gid=1000,mode=0700`